modules/Microsoft.Network/networkInterfaces/deploy.bicep

```diff
@@ -100,7 +100,7 @@ resource networkInterface 'Microsoft.Network/networkInterfaces@2021-08-01' = {
       id: networkSecurityGroupResourceId
     } : null
     ipConfigurations: [for (ipConfiguration, index) in ipConfigurations: {
-      name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
+      name: !empty(ipConfiguration.name) ? ipConfiguration.name : 'ipconfig-${index}'
```

Comment:
I'd like to suggest the following:
```bicep
contains(ipConfiguration, 'name') ? ipConfiguration.name : 'ipconfig0${index + 1}'
```
- This negates the need to provide `name: null` in the parameter file. Also, we don't have to provide the name as well so it can get a default.
- We can align to how the API defaults the naming for the first ipConfiguration, which is `ipconfig01`, hence incremental names would be 02, 03, .. 



So overall change would be something like this:

Reply:
Do you want to consider the rare case of more than 10 Ip configurations or are you ok with an Ip configuration such as ipconfig011?

Reply:
Yes I thought about that too. However, the likelihood of more than 9 IPs is very rare as you said, and I assume users to give them names to identify them as numbers are difficult to remember what IP does what.